src/Integration/YoastWooCommerceSeo.php

```diff
@@ -125,11 +127,38 @@ protected function get_identifier_value( string $key, WC_Product $product ) {
 		$product_id = $product->get_id();
 
 		if ( ! isset( $this->yoast_global_identifiers[ $product_id ] ) ) {
-			$product = $product instanceof WC_Product_Variation ? wc_get_product( $product->get_parent_id() ) : $product;
-
-			$this->yoast_global_identifiers[ $product_id ] = $product->get_meta( 'wpseo_global_identifier_values', true );
+			$this->yoast_global_identifiers[ $product_id ] = $this->get_identifier_meta( $product );
 		}
 
 		return ! empty( $this->yoast_global_identifiers[ $product_id ][ $key ] ) ? $this->yoast_global_identifiers[ $product_id ][ $key ] : null;
 	}
+
+	/**
+	 * Get identifier meta from product.
+	 * For variations fallback to parent product if meta is empty.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param WC_Product $product
+	 *
+	 * @return mixed|null
+	 */
+	protected function get_identifier_meta( WC_Product $product ) {
+		if ( ! $product ) {
+			return null;
+		}
+
+		if ( $product instanceof WC_Product_Variation ) {
+			$identifiers = $product->get_meta( 'wpseo_variation_global_identifiers_values', true );
+
+			if ( empty( $identifiers ) ) {
```

Comment:
`$identifiers` will never be empty as `$product->get_meta( 'wpseo_variation_global_identifiers_values', true );` will return an array with the format of `['gtin18' => '', 'gtin13' => '' ....]`.

We should check if all values are empty so we can use the parent's identifier.  

Suggested change:

```
if ( empty( array_filter( $identifiers ) ) ) {
 .....
}
```

Reply:
Thanks for the feedback, that makes sense. Although in cases where the variations haven't been re-saved after activating the extension $identifiers could be null. So we'll probably need a combination of both checks or some type casting otherwise array_filter will fail if we don't pass it an array.